### PR TITLE
Remove redundant type="text/javascript" from <script> elements

### PR DESCRIPTION
**Pull request by jdufresne**
In HTML5, `<script>` elements default to MIME type text/javascript. The HTML5 living standard and MDN recommend against including the attribute.

From https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-type

> The HTML5 specification urges authors to omit the attribute rather than provide a redundant MIME type.

From https://html.spec.whatwg.org/#the-script-element

> Authors should omit the type attribute instead of redundantly setting it.

### CODE REVIEW

**Review by tk0miya — COMMENTED** [review]
LGTM. Could you rebase this into 2.0 branch? Then I'll merge this soon.

**Comment by jdufresne** [comment]
Rebased :slightly_smiling_face:

File: sphinx/builders/html.py [hunk 1]
```diff
@@ -146,8 +146,6 @@ def __new__(cls, filename: str, **attributes: str) -> "JavaScript":
         self = str.__new__(cls, filename)  # type: ignore
         self.filename = filename
         self.attributes = attributes
-        self.attributes.setdefault('type', 'text/javascript')
-
```
Comment (tk0miya) [inline sphinx/builders/html.py:150]: trivial: This empty line is not related with this change. Please keep it as is.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits.

**Comment by jdufresne** [comment]
Restored blank line.

**Review by tk0miya — APPROVED** [review]
Thank you for quick fix!

**Comment by tk0miya** [comment]
Merged.
